docs_src/design/ucr/Extending-Device-Data.md

```diff
@@ -0,0 +1,78 @@
+The new analytics, utility, and north-bound microservices almost always need to add Device Resources to manage their configuration, transforms, and status reporting. These Resources are usually needed on a per-Device basis (rather than just overall service configuration or status), which can be seen as extending (adding to) the data of the original south-bound devices.
+
+Adding configuration and status via Resources that **extend** the original south-bound Device
+make this configuration and status data easily accessible and translatable to other Application Services and to the UI via REST;
+we think that this general solution is better than disparate solutions which add custom APIs in each Application Service to Get and Set this data.
+
+What is needed is a common means of showing the relationship between these added Resources, their owning service,
+and the original south-bound Device Resources; that is, to indicate that these Resources "extend" the original Device data.
+
+It is desirable that the means of conveying this information become standardized for those EdgeX microservices
+which provide and use it, hence proposing here that there be a common EdgeX way defined to do this.
+
+### Target Users
+- Product Developers
+- Device Owner
+- Device User
+- Device Maintainer
+- Cloud User
+- Service Provider
+- Software Integrator
+
+### Description
+Picture the extremely simple case of a south-bound sensor device that just measures Temperature and Humidity and provides these as Device Resources. If we then add analytics and north-bound microservices:
+- A Trending service that needs Device Resources to indicate that Temperature and Humidity are trended for, eg, Minimum, Average, and Maximum over a 1 hour trend interval.
+- An Alarming service that needs Device Resources to describe the Alarm Rules used to monitor Temperature and Humidity, plus a device-level InAlarm status.
+- A Cloud service that reports not just the Temperature and Humidity but also their Trend configuration and Alarm Rule Resources. In addition, the Cloud service adds its own Resources to direct the Cadence with which this Device's data is reported.
+
+Now scale this up to 100 such Temperature/Humidity sensors, and it grows difficult to match all of the free-standing 
+Resources to their original sensor data. And add the requirement that all these resources must be able to be seen 
+and managed locally via REST or Message Bus, and potentially from north-bound services like Modbus/TCP, and from 
+the Cloud (because everybody wants to control everything from the Cloud). 
+
+Furthermore, from the end user's point of view, the Trend configuration, Alarm Rules, and Cloud Cadence that are added for a given Device are all seen as aspects of the Temperature/Humidity Device, as is common 
+for Digital Twin representations, and not as separated, free-standing entities. 
+So there must be some means to relate the extended Device Resources to the original south-bound Device and its
```

Comment:
Presumably, multiple app services could want to add device resources to the original south-bound device data.  Should we put something in about collisions of device resource names?  I would assume that the device resource names must be unique across all services (device, app or otherwise)?  This will be an important requirement, because something would need to check before allowing a new resource to be added.

Reply:
Thanks, Jim. Good point; I'll add a requirement.